src/okd/get_version.sh

```diff
@@ -60,17 +102,25 @@ if [ "${OKD_XY}" = "latest" ]; then
 fi
 
 # Filter the version tags for the specified 'x.y' base version
-cat "${query_file}" | grep "^${OKD_XY}" | sort -V > "${version_file}" || true
+grep "^${OKD_XY}" "${query_file}" | sort -V > "${version_file}" || true
 if [ ! -s "${version_file}" ]; then
-    echo "ERROR: No OKD version tags found for the specified '${OKD_XY}' base version" >&2
+    echo "ERROR: No OKD version tag found for the '${OKD_XY}' version" >&2
     exit 1
 fi
 
-# Get the latest version tag giving priority to the released versions
-OKD_TAG="$(grep -Ev '\.rc\.|\.ec\.' "${version_file}" | tail -1 || true)"
-# If no released version tag is found, use the latest version tag
+# Try up to 3 times to get the latest version tag with both x86_64 and aarch64 release images available
+OKD_TAG=""
+for _ in {1..3}; do
+    cur_tag="$(pop_latest_version_tag "${version_file}")"
+    if [ "${NO_ARM64}" = "true" ] || check_arm64_release_image_exists "${cur_tag}" ; then
+        OKD_TAG="${cur_tag}"
+        break
+    fi
+done
```

Comment:
What I meant in DM was that we could have two flows which could be easier to understand because the logic flow would be more straightforward.

Instead of always taking AMD64 releases as a base, iterating over them to see if there's already a matching ARM64 release we could:

1. For the okd-release workflow - just take the latest AMD64
2. For any other workflow - just take the latest ARM64 release. Since we do the rebuild nightly, picking latest ARM64 means there is corresponding AMD64. We won't have to iterate over AMD builds to find one with ARM counterpart.